packages/typescript-estree/src/create-program/getParsedConfigFile.ts

```diff
@@ -0,0 +1,64 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import type * as ts from 'typescript/lib/tsserverlibrary';
+
+import { CORE_COMPILER_OPTIONS } from './shared';
+
+/**
+ * Utility offered by parser to help consumers parse a config file.
+ *
+ * @param configFile the path to the tsconfig.json file, relative to `projectDirectory`
+ * @param projectDirectory the project directory to use as the CWD, defaults to `process.cwd()`
+ */
+function getParsedConfigFile(
+  configFile: string,
+  projectDirectory?: string,
+): ts.ParsedCommandLine {
+  // We import this lazily to avoid its cost for users who don't use the service
+  // TODO: Once we drop support for TS<5.3 we can import from "typescript" directly
+  // eslint-disable-next-line @typescript-eslint/no-require-imports
+  const tsserver = require('typescript/lib/tsserverlibrary') as typeof ts;
```

Comment:
[Refactor] From the OP:

> We could try adding tsserver as a parameter to `getParsedConfigFile` to remove the extra lazy import if we think it will be a problem.

+1 from me, good idea. I think minimizing the number of ad-hoc `require(...)`s is very good.